### DISCUSSION TITLE
Making cloze cards

### DISCUSSION THREAD

**Discussion by debanjandhar12** [debanjandhar12/logseq-anki-sync#89] [Documentation]
Documentation has been moved from this location to this page: https://debanjandhar12.github.io/logseq-anki-sync/docs/card-types/cloze

**Comment by Vincenthetic** [comment]
**A kind reminder:** 
it seems that the `{{c2:: <information on the second card generated by the current note>}}` is, by default, reserved for `#reversed` cards, which will cause every `#incremental` note to generate a blank card (the second card). 
Therefore, it would be a good habit to use the `Empty Cards` functionality of the `Tools` menu of Anki to delete empty cards every time one creates an incremental card.

**Comment by Ripret** [comment]
Is there a way to use page tags inside of cloze and them still be searchable?

**Comment by Heavypilgrim** [comment]
Is it possible to create two cloze cards with some different gaps and some gaps that both have in common?

e.g. "The {{c1 earth}} is a {{c2 planet}}. Because {{c1&2 blabliblub}}."
==> the last gap should be asked/ hidden in both cards, c1 and c2.

**Reply by jaker-dotcom** [reply]
You could use a regex as in the example for the replacecloze block property above:

replacecloze:: " '/(earth|blabliblub)/gi', /(planet|blabliblub)/gi "

Edit: 
I am currently trying to figure out how exactly these work, but I didn't get it to work. Maybe it's not possible, maybe I did sth wrong. But the note certainly doesn't look right. somehow the word blabliblub got lost entirely!

**Comment by LightishRedd** [comment]
Anki's Cloze Macro Syntax with this plugin doesn't seem to support Markdown. For example, say I want to place a word wrapped in Markdown inside of `{{c1 **bold**}}`, the result in Logseq looks like this: `**bold**`. To keep the intended formatting in Logseq, I have to use the replaceCloze syntax, but that clutters up my notes in Logseq making them less readable.

So, I guess I have three questions. Is it possible to use Markdown syntax within Anki's Cloze Macro Syntax and I just overlooked it in the Documentation? If not, is it possible to implement it? Or is it possible to hide the replaceCloze syntax within HTML comments or something so that Logseq doesn't parse it?

**Reply by krvkir** [reply]
More to this, logseq links wrapped in double square brackets don't render when they're inside cloze deletions, and are not mentioned on backlinks of the referenced cards.

The best solution of course would be to properly handle markdown and links inside clozes, because hiding replacecloze properties would mean people will be forgetting to fix regexps after editing the block contents.

Actually, I don't care much about the formatting, but the linking is essential for me.

**Comment by jaker-dotcom** [comment]
It would be nice to be pointed to some resources that explain the replacecloze syntax. Because afaik there is a multitude of regex flavours (see [regex info](https://www.regular-expressions.info/tools.html)). And the trial and error strategy I am currently employing is buggling me. 
E.g. I am trying to create a cloze in math for all occurences of a_{13} except one. The one stands out by being the first in math mode and all the others follow an opening bracket. So I tried these options, but none seem to work:
"/(?<!$)a_{13}/g" "/(?<!\$)a_{13}/g" "/(?<$)a_{13}/g" "/(?<\()a_{13}/g" "/(?<()a_{13}/g"

Now I could ask such a question on here every time it arises, or teach myself how to figure it out / have a reference / know a program to to it for me instead.

**Comment by krvkir** [comment]
Hello and thank you for your work! It helps me a lot.

Is there a way to make cloze cards bidirectional? My use case is this: I read a book in English (which is not my native language), and there's a sentence with an unknown word, e.g. "The ship landed in a medley of noises.". I want to create a close like this:

The ship landed in a {{c1 medley::попурри}} of noises.

...and make two cards out of it: 1. show the translation, hide the English word; 2. show English word to make me remember the translation.

AFAIK there's a Reversed Cloze card type in Anki for this. Can I somehow specify the card type used for the block?

**Comment by LK0624** [comment]
![solution](https://github.com/user-attachments/assets/4bca6e53-ef3c-4772-ae0d-a10ce249bf1d)

alternative solutions 
both would do what you'd want.

**Reply by krvkir** [reply]
Requires duplication, but still quite a good workaround, thanks!

**Reply by LK0624** [reply]
The ship landed in a **{{c1 medley}} {{c2 попурри}}** of noises.

![CleanShot 2024-09-09 at 11 15 23@2x](https://github.com/user-attachments/assets/fad94fe9-8099-4880-b567-e78c405232f4)

![CleanShot 2024-09-09 at 11 15 34@2x](https://github.com/user-attachments/assets/28330d5b-8dab-453f-8d2b-7b16944a5cf9)

I think this could be better.
write in one block then C1, C2 
with bold or sth else like hint to just highlight what's being tested.
or you could just write a prompt like "translate to english or other native language" in the parent block for mentioning. 

then this is probably more easy and clean

**Reply by wadid** [reply]
@krvkir @LK0624 do you know a way to create the clozes anki like with hotkey and automatic numbering?

**Reply by krvkir** [reply]
Nope, I insert curly brackets and cX identifiers manually. A hotkey which wraps a selection with cloze markup with a proper number would be a nice thing to have, but honestly it'll save only a fraction of second for me.

**Reply by krvkir** [reply]
Actually, the most time consuming operation in my pipeline is fetching (copy-pasting) excerpts from an epub book. (I read books in Calibre, highlight new words and write translations in each highlight's comment.)  I lazily think about writing a script which would fetch those highlights from Calibre highlights file (it's just an xml) and produce ready-to-paste pieces for Logseq, which'll greatly reduce my need for manual cloze markup.

**Comment by wadid** [comment]
Is the highlight function something you implemented or is it native from
Calibre?

Am Di., 8. Okt. 2024 um 20:35 Uhr schrieb Kirill Krasnoschekov <
***@***.***>:

**Reply by krvkir** [reply]
It's a native feature in Calibre's internal book reader. You just double-click on a word or select a piece of text, and the tooltip shows up. You can even set up a translator site which will show up in the side panel of the reader and translate the selected excerpt.

**Comment by BillGatesPriv** [comment]
How do you write `{{c1 $\frac{1}{\sqrt{2}}$ }}` with cloze syntax?

In Logseq it works fine, however anki finds the closest double closing brackets, breaking the formula.

**Reply by BillGatesPriv** [reply]
adding a space between the brackets `}` fixes the problem :)
